### PR TITLE
qemu.test: qemu_img_luks_key_management

### PR DESCRIPTION
**Pull request by XueqiangWei**
1. add a new passowrd to free keyslot
2. erase password from keyslot

Signed-off-by: Xueqiang Wei <xuwei@redhat.com>
ID: 1945872

### CODE REVIEW

**Comment by XueqiangWei** [comment]
depends on avocado-framework/avocado-vt/pull/2989

**Comment by XueqiangWei** [comment]
(1/7) Host_RHEL.m8.u4.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_luks_key_management.add_new_password_to_keyslot.with_specified_index.q35: PASS (15.92 s)
 (2/7) Host_RHEL.m8.u4.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_luks_key_management.add_new_password_to_keyslot.without_index.q35: PASS (15.70 s)
 (3/7) Host_RHEL.m8.u4.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_luks_key_management.add_new_password_to_keyslot.with_active_keyslot.q35: PASS (9.95 s)
 (4/7) Host_RHEL.m8.u4.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_luks_key_management.add_new_password_to_keyslot.with_invalid_keyslot.q35: PASS (9.93 s)
 (5/7) Host_RHEL.m8.u4.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_luks_key_management.erase_password_from_keyslot.with_specified_index.q35: PASS (18.21 s)
 (6/7) Host_RHEL.m8.u4.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_luks_key_management.erase_password_from_keyslot.with_specified_password.q35: PASS (22.61 s)
 (7/7) Host_RHEL.m8.u4.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_luks_key_management.erase_password_from_keyslot.with_only_active_keyslot.q35: PASS (9.90 s)

**Comment by XueqiangWei** [comment]
@zhencliu Please help review it. Many thanks.

File: qemu/tests/qemu_img_luks_key_management.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    if err_info:
+        if not re.search(err_info, cmd_result.stderr.decode(), re.I):
+            test.fail("Failed to get error information. The actual error "
+                      "information is %s." % cmd_result.stderr.decode())
+    elif cmd_result.exit_status != 0:
+        test.fail("Failed to amend image %s. The error information is "
+                  "%s." % (stg_img.image_filename, cmd_result.stderr.decode()))
+    else:
+        info = json.loads(stg_img.info(output="json"))
+        if stg_img.image_format == "qcow2":
```
Comment (zhencliu) [inline qemu/tests/qemu_img_luks_key_management.py:83]: have we ever covered this scenario, encrypted qcow2? It looks like that you use 'only luks' filter in cfg

File: qemu/tests/qemu_img_luks_key_management.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    6. (1) Erase password from keyslot by giving a keyslot index
+       in this case, adding a password to keyslot 7 and then erase it
+       e.g. qemu-img amend --object secret,id=sec0,data=redhat
+            --object secret,id=sec1,data=amend
+            -o state=inactive,keyslot=7
+            'json:{"file": {"driver": "file", "filename": "base.luks"},
+            "driver": "luks", "key-secret": "sec0"}'
+       (2) check slots -> active is False
+    7. (1) Erase password from keyslot by giving the password
+       in this case, adding a password to keyslot 7 and then erase it
```
Comment (zhencliu) [inline qemu/tests/qemu_img_luks_key_management.py:38]: adding a password to a free keyslot, and then erase it by password? or object id?

File: qemu/tests/qemu_img_luks_key_management.py (hunk 1 above)
Comment (zhencliu) [reply qemu/tests/qemu_img_luks_key_management.py:83]: I see that you put qcow2 with encryption in another patch.

Comment (zhencliu) [reply qemu/tests/qemu_img_luks_key_management.py:83]: Then can we merge the two cfgs into one? It looks most conf are the same.

e.g. 

only luks qcow2
luks:
    keyslot =7
qcow2:
     amend_encrypt.keyslot = 7

Comment (XueqiangWei) [reply qemu/tests/qemu_img_luks_key_management.py:83]: There are many different parameters between them. So I sent two patches.  Merge the two cfgs into one, I think it's OK, Just the cfg is too long.
For luks:
image_format_stg = luks
image_secret_stg = redhat
amend_state
amend_new-secret
amend_keyslot
amend_old-secret

For luks-inside-qcow2:
image_format_stg = qcow2
image_secret_stg = redhat
image_encryption_stg = luks
amend_encrypt.state
amend_encrypt.new-secret
amend_encrypt.keyslot
amend_encrypt.old-secret

only luks qcow2
image_format_stg = luks
image_secret_stg = redhat
qcow2:
        image_format_stg = qcow2
        image_encryption_stg = luks

amend_keyslot = 1
qcow2:
        amend_encrypt.keyslot = 1
......

File: qemu/tests/qemu_img_luks_key_management.py (hunk 2 above)
Comment (XueqiangWei) [reply qemu/tests/qemu_img_luks_key_management.py:38]: Tested with two way (1) erase it with_specified_index (2) erase it with_specified_password, I will update the doc string. Thanks.

File: qemu/tests/qemu_img_luks_key_management.py (hunk 1 above)
Comment (zhencliu) [reply qemu/tests/qemu_img_luks_key_management.py:83]: OK,that makes sense, it's OK for me to split one to two cfg

**Comment by zhencliu** [comment]
LGTM

Let's wait till vt patch is available

**Comment by XueqiangWei** [comment]
(1/7) Host_RHEL.m8.u4.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_luks_key_management.add_new_password_to_keyslot.with_specified_index.q35: PASS (15.86 s)
 (2/7) Host_RHEL.m8.u4.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_luks_key_management.add_new_password_to_keyslot.without_index.q35: PASS (15.67 s)
 (3/7) Host_RHEL.m8.u4.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_luks_key_management.add_new_password_to_keyslot.with_active_keyslot.q35: PASS (9.91 s)
 (4/7) Host_RHEL.m8.u4.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_luks_key_management.add_new_password_to_keyslot.with_invalid_keyslot.q35: PASS (9.92 s)
 (5/7) Host_RHEL.m8.u4.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_luks_key_management.erase_password_from_keyslot.with_specified_index.q35: PASS (18.05 s)
 (6/7) Host_RHEL.m8.u4.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_luks_key_management.erase_password_from_keyslot.with_specified_password.q35: PASS (22.58 s)
 (7/7) Host_RHEL.m8.u4.product_av.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.x86_64.io-github-autotest-qemu.qemu_img_luks_key_management.erase_password_from_keyslot.with_only_active_keyslot.q35: PASS (9.91 s)

**Comment by XueqiangWei** [comment]
@zhencliu Added the prefix to secret_id and secret_data. Please help review it again. Many thanks.

**Comment by zhencliu** [comment]
VT patch 2736 is to be merged soon, so merge this patch

**Pull request merged**